Source/missiles.cpp

```diff
@@ -3278,7 +3264,7 @@ void MI_LightningArrow(int i)
 					AddMissile(
 					    missile.position.tile,
 					    missile.position.start,
-					    i,
+					    missile._mimfnum,

```

Comment:
Are you sure it's safe to remove this hack?

Reply:
I have fixed this hack in another PR related to cleaning up  `AddMissile(misdir)`: https://github.com/diasurgical/devilutionX/pull/2692

I'm working on splitting out a function for handling parent spells, which you can use to fix the issue with `MI_Firebolt` as well.

Reply:
@AJenbo thank you for addressing this issue in your pull request! I will rebase after your pull request gets merged to master branch.